src/RadioGroup/RadioGroup.tsx

```diff
@@ -52,6 +52,12 @@ export const RadioGroup = <Value extends BaseValueType>({
   )
 }
 
+export const RadioGroup = forwardRef(RadioGroupComponent) as <
+  Value extends BaseValueType = BaseValueType,
+>(
+  p: RadioGroupProps<Value> & React.RefAttributes<HTMLInputElement>,
+) => ReactElement | null
```

Comment:
`forwardRef` (or HOC in general) doesn't play nicely with generic
This seems like the best approach (that I could find at least)